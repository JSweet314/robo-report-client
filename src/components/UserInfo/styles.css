.user-info {
  height: 100%;
  min-height: 50vh;
  max-width: 70%;
  margin: 2rem auto;
  font-size: 2rem;
}

.user-info__p {
  line-height: 1.3;
}

.user-info__button {
  font-weight: bold;
  width: fit-content;
  margin: 1.5rem 1rem 0 0;
  padding: .2rem;
  border: 1px solid darkgray;
  border-radius: 10px;
  padding: .5rem;
  margin: 0;
  float: right;
}

.user-info__button--edit {
  margin: 0 2rem 0 0;
}