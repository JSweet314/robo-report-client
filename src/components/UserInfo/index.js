import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const UserInfo = ({ user, handleOAuthSignOut, toggleEdit }) => {
  return (
    <div className="user-info">
      <button
        className="user-info__button user-info__button--edit"
        onClick={toggleEdit}
      >
        Edit My Information
      </button>
      <p className="user-info__p">
        <strong>First Name: </strong>
        <span>{user.firstName}</span>
      </p>
      <p className="user-info__p">
        <strong>Last Name: </strong>
        <span>{user.lastName}</span>
      </p>
      <p className="user-info__p">
        <strong>Email: </strong>
        <span>{user.email}</span>
      </p>
      <p className="user-info__p">
        <strong>Phone: </strong>
        <span>{user.phone}</span>
      </p>
      <p className="user-info__p">
        <strong>Phone Type: </strong>
        <span>{user.phoneType}</span>
      </p>
      <p className="user-info__p">
        <strong>Phone Location: </strong>
        <span>{user.phoneLocation}</span>
      </p>
      <p className="user-info__p">
        <strong>Address: </strong>
        <span>{user.address}</span>
      </p>
      <p className="user-info__p">
        <strong>State: </strong>
        <span>{user.state}</span>
      </p>
      <p className="user-info__p">
        <strong>City: </strong>
        <span>{user.city}</span>
      </p>
      <p className="user-info__p">
        <strong>Zip Code: </strong>
        <span>{user.zipcode}</span>
      </p>
      <button className="user-info__button" onClick={handleOAuthSignOut}>
        Sign Out
      </button>
    </div>
  );
};

UserInfo.propTypes = {
  user: PropTypes.object.isRequired,
  handleOAuthSignOut: PropTypes.func.isRequired,
  toggleEdit: PropTypes.func.isRequired
};

export default UserInfo;
