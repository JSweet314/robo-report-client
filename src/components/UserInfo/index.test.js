import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from './';

describe('UserInfo', () => {
  let wrapper;
  const mockHandleOAuthSignOut = jest.fn();
  const mockToggleEdit = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <UserInfo
        user={{ firstName: 'Jerry', phone: '111-222-3333' }}
        handleOAuthSignOut={mockHandleOAuthSignOut}
        toggleEdit={mockToggleEdit}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
