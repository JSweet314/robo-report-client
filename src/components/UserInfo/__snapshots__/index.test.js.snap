// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo should match the snapshot 1`] = `
<div
  className="user-info"
>
  <button
    className="user-info__button"
    onClick={[Function]}
  >
    Sign Out
  </button>
  <button
    className="user-info__button user-info__button--edit"
    onClick={[Function]}
  >
    Edit My Information
  </button>
  <p
    className="user-info__p"
  >
    <strong>
      First Name: 
    </strong>
    <span>
      Jerry
    </span>
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      Last Name: 
    </strong>
    <span />
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      Email: 
    </strong>
    <span />
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      Phone: 
    </strong>
    <span>
      111-222-3333
    </span>
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      Phone Type: 
    </strong>
    <span />
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      Phone Location: 
    </strong>
    <span />
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      Address: 
    </strong>
    <span />
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      State: 
    </strong>
    <span />
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      City: 
    </strong>
    <span />
  </p>
  <p
    className="user-info__p"
  >
    <strong>
      Zip Code: 
    </strong>
    <span />
  </p>
</div>
`;
